Actions/DetermineSecrets/action.yaml

```diff
@@ -0,0 +1,130 @@
+name: Determine Secrets
+author: Microsoft Corporation
+inputs:
+  shell:
+    description: Shell in which you want to run the action (powershell or pwsh)
+    required: false
+    default: powershell
+  getSecrets:
+    description: Comma-separated list of Secrets to get. Secrets preceded by an asterisk are returned encrypted
+    required: true
+  useGhTokenWorkflowForPush:
+    description: Determines whether you want to use the GhTokenWorkflow secret for TokenForPush
+    required: false
+    default: 'false'
+outputs:
+  FORMATSTR:
```

Comment:
I still cannot understand why the action doesn't simply output a string that represents comma-separated secret names, that need to be read.
Why does the formatting have to be on the action caller instead of here?

Reply:
The challenge is to resolve all GitHub secrets in yaml.
Before this, we transferred all GitHub Secrets to PowerShell and had PowerShell do the working (i.e. ToJson(Secrets))
I tried a lot of different things in a prototype (here: https://github.com/freddydk/test/blob/main/.github/workflows/Secrets.yaml) and ended up with this construct as the only way I could make that work.
If the determineSecrets returns a list of secret names - how would you then get the secrets in YAML in a generic matter?

Reply:
I finally get the limitation. Your approach works, so let's go with it.